diesel_cli/src/main.rs

```diff
@@ -186,6 +188,59 @@ fn convert_absolute_path_to_relative(target_path: &Path, mut current_path: &Path
     result.join(target_path.strip_prefix(current_path).unwrap())
 }
 
+fn run_infer_schema(matches: &ArgMatches) {
+    let database_url = database::database_url(matches);
+    let schema_name = None;
+    let schema = diesel_infer_schema::infer_schema_for_schema_name(&database_url, schema_name);
+    let mut out = String::new();
+    let mut indent = String::new();
+    let mut skip_space = false;
+    let mut last_char = ' ';
+    
+
+    for c in schema.as_str().chars() {
```

Comment:
The problem with rustfmt is that it fails on formating the macros.
Running rustfmt on the output of `quote::Token::as_str()` will produce something like

```
mod infer_users {
    table ! { users ( id ) { id -> :: diesel :: types :: Int4 , username -> :: diesel :: types :: Varchar , password -> :: diesel :: types :: Varchar , access_level -> :: diesel :: types :: Int2 , } }
}
;
```

In my opinion this is far from optimal. (There seems to be a rustfmt [issue](https://github.com/rust-lang-nursery/rustfmt/issues/438)) 

If rustfmt would work on the generated code, running rustfmt would be the better solution. So I'm not realy sure how to handle this.